### PR TITLE
Add okta_device table

### PR DESCRIPTION
**Pull request by misraved**
# Example query results
<details>
  <summary>Results</summary>
  
```
> select * from okta_device where id = 'guocxyyla1oe84w2p697'
+------------------+----------------------+---------------------------+---------------------------+----------------------+---------------+--------+---------------------------------------------------------------------------------------->
| display_name     | id                   | created                   | last_updated              | resource_id          | resource_type | status | profile                                                                                >
+------------------+----------------------+---------------------------+---------------------------+----------------------+---------------+--------+---------------------------------------------------------------------------------------->
| Redmi Note 7 Pro | guocxyyla1oe84w2p697 | 2024-04-03T13:29:01+05:30 | 2024-04-03T13:29:01+05:30 | guocxyyla1oe84w2p697 | UDDevice      | ACTIVE | {"diskEncryptionType":"USER","displayName":"Redmi Note 7 Pro","manufacturer":"Xiaomi",">
+------------------+----------------------+---------------------------+---------------------------+----------------------+---------------+--------+---------------------------------------------------------------------------------------->
> select * from okta_device
+------------------+----------------------+---------------------------+---------------------------+----------------------+---------------+--------+---------------------------------------------------------------------------------------->
| display_name     | id                   | created                   | last_updated              | resource_id          | resource_type | status | profile                                                                                >
+------------------+----------------------+---------------------------+---------------------------+----------------------+---------------+--------+---------------------------------------------------------------------------------------->
| Redmi Note 7 Pro | guocxyyla1oe84w2p697 | 2024-04-03T13:29:01+05:30 | 2024-04-03T13:29:01+05:30 | guocxyyla1oe84w2p697 | UDDevice      | ACTIVE | {"diskEncryptionType":"USER","displayName":"Redmi Note 7 Pro","manufacturer":"Xiaomi",">
+------------------+----------------------+---------------------------+---------------------------+----------------------+---------------+--------+---------------------------------------------------------------------------------------->

>  select
  display_name,
  id,
  profile ->> 'DiskEncryptionType' as disk_encryption_type,
  profile ->> 'DisplayName' as DisplayName,
  profile ->> 'Imei' as Imei,
  profile ->> 'IntegrityJailbreak' as IntegrityJailbreak,
  profile ->> 'Manufacturer' as Manufacturer,
  profile ->> 'Meid' as Meid,
  profile ->> 'Model' as Model,
  profile ->> 'OsVersion' as OsVersion,
  profile ->> 'Platform' as Platform,
  profile ->> 'Registered' as Registered,
  profile ->> 'SecureHardwarePresent' as SecureHardwarePresent,
  profile ->> 'SerialNumber' as SerialNumber,
  profile ->> 'Sid' as Sid,
  profile ->> 'TpmPublicKeyHash' as TpmPublicKeyHash
from
  okta_device;
+------------------+----------------------+----------------------+-------------+--------+--------------------+--------------+--------+--------+-----------+----------+------------+-----------------------+--------------+--------+-------->
| display_name     | id                   | disk_encryption_type | displayname | imei   | integrityjailbreak | manufacturer | meid   | model  | osversion | platform | registered | securehardwarepresent | serialnumber | sid    | tpmpubl>
+------------------+----------------------+----------------------+-------------+--------+--------------------+--------------+--------+--------+-----------+----------+------------+-----------------------+--------------+--------+-------->
| Redmi Note 7 Pro | guocxyyla1oe84w2p697 | <null>               | <null>      | <null> | <null>             | <null>       | <null> | <null> | <null>    | <null>   | <null>     | <null>                | <null>       | <null> | <null> >
+------------------+----------------------+----------------------+-------------+--------+--------------------+--------------+--------+--------+-----------+----------+------------+-----------------------+--------------+--------+-------->

```
</details>

### CODE REVIEW

File: okta/table_okta_device.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+//// HYDRATE FUNCTIONS
+
+func getOktaDevice(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+
+	var deviceId string
+	if h.Item != nil {
+		deviceId = *h.Item.(okta.DeviceList).Id
```
Comment (bigdatasourav) [inline okta/table_okta_device.go:125]: is there any column that specifically depends on the getOktaDevice function?

Comment (ParthaI) [reply okta/table_okta_device.go]: Add, my bad. Updated the function getOktaDevice

File: okta/connect.go [hunk 2]
```diff
@@ -128,3 +129,83 @@ func Connect(ctx context.Context, d *plugin.QueryData) (*okta.Client, error) {
 
 	return client, err
 }
+
+
+func ConnectV4(ctx context.Context, d *plugin.QueryData) (*oktaV4.APIClient, error) {
```
Comment (bigdatasourav) [inline okta/connect.go:134]: Can we use the existing Connect function and update it with oktav4?

File: okta/table_okta_device.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getOktaDevice(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+
+	deviceId := d.EqualsQuals["id"].GetStringValue()
```
Comment (bigdatasourav) [inline okta/table_okta_device.go:123]:
```suggestion
	deviceId := d.EqualsQualString("id")
```

File: okta/connect.go (hunk 2 above)
Comment (ParthaI) [reply okta/connect.go:134]: @bigdatasourav, I believe that would be a separate PR, for migrating the V1 API to V4. Should we do it in this PR?

**Comment by mattjurczyk** [comment]
Is it possible to add https://developer.okta.com/docs/api/openapi/okta-management/management/tag/Device/#tag/Device/operation/listDevices!in=query&path=expand&t=request to the api call? this will help correlate devices to users.

**Comment by mattjurczyk** [comment]
`deviceReq := client.DeviceAPI.ListDevices(ctx).Expand("userSummary")` does the trick

**Review by mattjurczyk — APPROVED** [review]

**Review by bigdatasourav — APPROVED** [review]

**Comment by mattjurczyk** [comment]
@bigdatasourav was this ever added to the plugin? I dont see okta devices in the docs

**Comment by ParthaI** [comment]
Hi @mattjurczyk, apologies for the delay in the release.

We have added support for `okta_device` in [this PR](https://github.com/turbot/steampipe-plugin-okta/pull/113). However, we might have missed releasing the plugin.

@misraved, could you please take a look and release the plugin?

**Pull request merged**